MarsRover.Tests/ControllerShould.cs

```diff
@@ -70,20 +70,20 @@ public void Run_ReturnsRoverReport_GivenRoverCommandsThatResultInAnObstacleAnd2C
         }
 
         [Fact]
-        public void Run_ReturnsEndMessage_Given2SetsOfRoverCommands()
+        public void Run_ReturnsRoverEndLocation_GivenMapAnd2SetsOfRoverCommands()
         {
-            _mockInput.SetupSequence(_ => _.ReadLine())
+            _mockInput.SetupSequence(i => i.ReadLine())
                 .Returns("TestFiles/validFile1.txt")
                 .Returns("N")
                 .Returns("2,0")
                 .Returns("lff")
                 .Returns("rff")
                 .Returns("q");
-            var expectedString = "Mars Rover has completed his mission.";
+            var expectedLocation = new Location(1,1);
 
             _controller.Run();
 
-            _output.GetLastOutput().Should().BeEquivalentTo(expectedString);
+            _controller.Rover.Location.Should().BeEquivalentTo(expectedLocation);

```

Comment:
Could consider using your last map output function here to test “behaviour” of the controller i.e. outputs rather than the Rover’s state

Reply:
Just updated